src/core/Global.ttslua

```diff
@@ -1035,3 +1039,92 @@ function titleSplash(scenarioName)
     soundCubeApi.playSoundByName("Deep Bell")
   end
 end
+
+---------------------------------------------------------
+-- Update notification related functionality
+---------------------------------------------------------
+
+-- grabs the latest mod version and release notes from GitHub (called onLoad())
+function getModVersion()
+  WebRequest.get(SOURCE_REPO .. '/modversion.json', compareVersion)
+end
+
+-- compares the modversion with GitHub and possibly shows the update notification
+function compareVersion(request)
+  if request.is_error then
+    log(request.error)
+    return
+  end
+
+  -- global variable to make it accessible for other functions
+  modMeta = JSON.decode(request.text)
+
+  -- stop here if on latest version or "don't show again" was clicked
+  if MOD_VERSION == modMeta["latestVersion"] then return end
+  
+  if dontNotify[modMeta["latestVersion"]] then return end
+
+  -- end here if no release notes are found
+  if modMeta["releaseNotes"][modMeta["latestVersion"]] == nil then
+    log("Release notes for latest version not found!")
+  else
+    updateNotificationLoading()
+
+    -- delay to avoid lagging during onLoad()
+    Wait.time(function() UI.show("FinnIcon") end, 1)
+  end
+end
+
+-- updates the XML update notification based on the mod metadata
+function updateNotificationLoading()
+  -- grab data
+  local highlights = modMeta["releaseNotes"][modMeta["latestVersion"]]["Highlights"]
+  local ui = UI.getXmlTable()
+
+  -- update the header
+  local header = find_tag_with_id(ui, 'notificationHeader')
+  header.value = header.value .. modMeta["latestVersion"]
```

Comment:
Any advantage to compositing the header string like this?  It splits the text value (some here, some XML), and while I can't find a path to make it happen runs the risk of this triggering multiple times so you'd get something like New version available: 3.1.03.1.0.  Simpler and safer to just set the full string here.